File: CS/GridGroupSelect/SampleDataRow.cs
using System;
using System.Collections.Generic;
using System.Collections.ObjectModel;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using System.Windows;

namespace dxSample {
    class SampleDataRow : DependencyObject {
        public int Id {
            get { return (int)GetValue(IdProperty); }
            set { SetValue(IdProperty, value); }
        }
        public string Group {
            get { return (string)GetValue(GroupProperty); }
            set { SetValue(GroupProperty, value); }
        }
        public string Name {
            get { return (string)GetValue(NameProperty); }
            set { SetValue(NameProperty, value); }
        }
        public bool HasFlag {
            get { return (bool)GetValue(HasFlagProperty); }
            set { SetValue(HasFlagProperty, value); }
        }

        public static readonly DependencyProperty HasFlagProperty = DependencyProperty.Register("HasFlag", typeof(bool), typeof(SampleDataRow), new PropertyMetadata(false));
        public static readonly DependencyProperty NameProperty = DependencyProperty.Register("Name", typeof(string), typeof(SampleDataRow), new PropertyMetadata(""));
        public static readonly DependencyProperty GroupProperty = DependencyProperty.Register("Group", typeof(string), typeof(SampleDataRow), new PropertyMetadata(""));
        public static readonly DependencyProperty IdProperty = DependencyProperty.Register("Id", typeof(int), typeof(SampleDataRow), new PropertyMetadata(0));

        public static IList<SampleDataRow> CreateRows() {
            IList<SampleDataRow> result = new List<SampleDataRow>();
            for(int i = 0; i < 100; i++) {
                result.Add(new SampleDataRow() {
                    Id = i,
                    Group = String.Format("group {0}", i % 10),
                    Name = String.Format("name {0}", i % 3),
                    HasFlag = (i % 3 == 0),
                });
            }
            return result;
        }
    }
}

File: VB/GridGroupSelect/SampleDataRow.vb
Imports System.Collections.Generic
Imports System.Windows

Namespace dxSample

    Friend Class SampleDataRow
        Inherits DependencyObject

        Public Property Id As Integer
            Get
                Return CInt(GetValue(IdProperty))
            End Get

            Set(ByVal value As Integer)
                SetValue(IdProperty, value)
            End Set
        End Property

        Public Property Group As String
            Get
                Return CStr(GetValue(GroupProperty))
            End Get

            Set(ByVal value As String)
                SetValue(GroupProperty, value)
            End Set
        End Property

        Public Property Name As String
            Get
                Return CStr(GetValue(NameProperty))
            End Get

            Set(ByVal value As String)
                SetValue(NameProperty, value)
            End Set
        End Property

        Public Property HasFlag As Boolean
            Get
                Return CBool(GetValue(HasFlagProperty))
            End Get

            Set(ByVal value As Boolean)
                SetValue(HasFlagProperty, value)
            End Set
        End Property

        Public Shared ReadOnly HasFlagProperty As DependencyProperty = DependencyProperty.Register("HasFlag", GetType(Boolean), GetType(SampleDataRow), New PropertyMetadata(False))

        Public Shared ReadOnly NameProperty As DependencyProperty = DependencyProperty.Register("Name", GetType(String), GetType(SampleDataRow), New PropertyMetadata(""))

        Public Shared ReadOnly GroupProperty As DependencyProperty = DependencyProperty.Register("Group", GetType(String), GetType(SampleDataRow), New PropertyMetadata(""))

        Public Shared ReadOnly IdProperty As DependencyProperty = DependencyProperty.Register("Id", GetType(Integer), GetType(SampleDataRow), New PropertyMetadata(0))

        Public Shared Function CreateRows() As IList(Of SampleDataRow)
            Dim result As IList(Of SampleDataRow) = New List(Of SampleDataRow)()
            For i As Integer = 0 To 100 - 1
                result.Add(New SampleDataRow() With {.Id = i, .Group = String.Format("group {0}", i Mod 10), .Name = String.Format("name {0}", i Mod 3), .HasFlag = i Mod 3 = 0})
            Next

            Return result
        End Function
    End Class
End Namespace
